src/Microsoft.Dnx.Runtime/Internal/ConcurrencyUtilities.cs

```diff
@@ -10,6 +11,11 @@ namespace Microsoft.Dnx.Runtime.Internal
 {
     public static class ConcurrencyUtilities
     {
+#if DNXCORE50
+        private static ConcurrentDictionary<string, System.Threading.Semaphore> _nameSemaphore =
```

Comment:
Also, how big can this get? This for every file path touched during restore right? Have we tested MVC after deleting `.dnx\packages`? We should remove the entry in dispose right?